### ISSUE TITLE
Doxygen not in path -> more explicit message?

### ISSUE THREAD

**Issue by adrienbernede** [LLNL/blt#348]
Doxygen wasn’t in my path... I think we could make the error message more explicit

```
CMake Error at cmake/blt/cmake/SetupDocs.cmake:36 (add_dependencies):
  Cannot add target-level dependencies to non-existent target "doxygen_docs".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.
Call Stack (most recent call first):
  src/docs/doxygen/CMakeLists.txt:7 (blt_add_doxygen_target)
```

1. How about making find doxygen _required_?

https://github.com/LLNL/blt/blob/e5564c491cb9032e529768ffc1e7b2696659da92/cmake/thirdparty/SetupThirdParty.cmake#L72

2. Or testing `DOXYGEN_FOUND` when entering the `blt_add_doxygen_target`.

https://github.com/LLNL/blt/blob/e5564c491cb9032e529768ffc1e7b2696659da92/cmake/SetupDocs.cmake#L27

IMO, 1. is legitimate here. But I may be wrong...

**Comment by cyrush** [comment]
common pattern is to guard any of your `blt_add_doxygen_target()` commands with `DOXYGEN_FOUND` in  your CMake -- it would be good to add a sanity check in `blt_add_doxygen_target()` as well that gives helpful advice.

**Comment by cyrush** [comment]
I don't think we want to require doxygen, as not all users of BLT are using doxygen -- also folks want to be able to turn off docs

**Comment by adrienbernede** [comment]
OK, I think I got it:

1. BLT allows to (de)activate doxygen with `ENABLE_DOXYGEN` which is set to `ON` by default.
2. The project has to ensure that doxygen is found and protect calls to `blt_add_doxygen_target` with `DOXYGEN_FOUND`.

However, if I do that, I will not be warned that I asked for doxygen documentation, and it was skipped because it is not in my `PATH`.

Also, BLT [documentation](https://llnl-blt.readthedocs.io/en/latest/creating_documentation.html#calc-pi-doxygen-example) says that:
> *blt_add_doxygen_target*
A macro to create a named doxygen target for API documentation. Assumes there is a Doxyfile.in doxygen configuration file in the current directory. _This macro is active when BLT is configured with a valid DOXYGEN_EXECUTABLE path._

However my macro was active even in `DOXYGEN_EXECUTABLE` was not available.

What about this alternate pattern:

1. In project, protect calls to `blt_add_doxygen_target` with `ENABLE_DOXYGEN`
2. In BLT, ensure that when `blt_add_doxygen_target` is called, `DOXYGEN_FOUND` is true.

If that sounds right it volunteer to make the changes in BLT and documentation.
